test/micro/org/openjdk/bench/java/math/Fp16ConversionBenchmark.java

```diff
@@ -0,0 +1,96 @@
+//
+// Copyright (c) 2003, 2022, Oracle and/or its affiliates. All rights reserved.
+// DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+//
+// This code is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License version 2 only, as
+// published by the Free Software Foundation.
+//
+// This code is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// version 2 for more details (a copy is included in the LICENSE file that
+// accompanied this code).
+//
+// You should have received a copy of the GNU General Public License version
+// 2 along with this work; if not, write to the Free Software Foundation,
+// Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+// Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+// or visit www.oracle.com if you need additional information or have any
+// questions.
+//
+//
+package org.openjdk.bench.java.math;
+
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.*;
+
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Thread)
+@Warmup(iterations = 5, time = 1)
+@Measurement(iterations = 5, time = 1)
+@Fork(value = 3)
+public class Fp16ConversionBenchmark {
+
+  @Param({"2048"})
+  public int TESTSIZE;
+
+  public short[] HFargV1;
+  public short[] ResHF;
+  public float[] FargV1;
+  public float[] ResF;
+
+  public final short[] HFspecialVals = {
+      0, (short)0xffff, (short)0xfc00, 0x7c00, 0x7bff, 0x400
+  };
+
+  public final float[] FspecialVals = {
+      0.0f, Float.NaN, Float.NEGATIVE_INFINITY, Float.POSITIVE_INFINITY,
+      Float.MAX_VALUE, Float.MIN_VALUE
+  };
+
+  @Setup(Level.Trial)
+  public void BmSetup() {
+      int i = 0;
+      Random r = new Random(1024);
+
+      HFargV1 = new short[TESTSIZE];
+      ResHF = new short[TESTSIZE];
+
+      for (; i < HFspecialVals.length; i++) {
+          HFargV1[i] = HFspecialVals[i];
+      }
+
+      for (; i < TESTSIZE; i++) {
+          HFargV1[i] = Float.floatToFloat16(r.nextFloat());;
+      }
+
+      FargV1 = new float[TESTSIZE];
+      ResF = new float[TESTSIZE];
+
+      i = 0;
+      for (; i < FspecialVals.length; i++) {
+          FargV1[i] = FspecialVals[i];
+      }
+
+      for (; i < TESTSIZE; i++) {
+          FargV1[i] = Float.float16ToFloat((short)r.nextInt());
+      }
+  }
+
+  @Benchmark
+  public void floatToFloat16() {
+      for (int i = 0; i < TESTSIZE; i++) {
```

Comment:
Use the array length instead, it's more idiomatic e.g.:
```suggestion
      for (int i = 0; i < FargV1.length; i++) {
```